tests/integration/test_s3.py

```diff
@@ -1033,5 +1034,54 @@ def setUp(self):
         self.client = self.create_client()
 
 
-if __name__ == '__main__':
-    unittest.main()
+class TestRegionRedirect(BaseS3ClientTest):
+    def setUp(self):
+        super(TestRegionRedirect, self).setUp()
+        self.bucket_region = 'eu-central-1'
+        self.client_region = 'us-west-2'
+
+        self.client = self.session.create_client(
+            's3', region_name=self.client_region,
+            config=Config(signature_version='s3v4'))
+
+    def create_bucket(self, region_name, bucket_name=None):
+        saved_client = self.client
+        if region_name != self.client.meta.region_name:
```

Comment:
Seems like it's cleaner if you just allow create_bucket to be passed a client to use.